include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -146,8 +146,317 @@ namespace Differentiation
        */
       bool
       is_valid_substitution_symbol(const SymEngine::Basic &entry);
+
+      /**
+       * A convenience function to set the @p value associated with
+       * the @p symbol in the @p substitution_map.
+       *
+       * Using this function ensures that the @p symbol is one that is
+       * valid specifically for the purpose of symbolic substitution.
+       * It must therefore represent a symbol or symbolic derivative,
+       * otherwise an error will be thrown.
+       */
+      void
+      set_value_in_symbol_map(
+        types::substitution_map &                     substitution_map,
+        const SymEngine::RCP<const SymEngine::Basic> &symbol,
+        const SymEngine::RCP<const SymEngine::Basic> &value);
     } // namespace internal
 
+    /**
+     * Return a symbolic map that has the entry key given by @p symbol.
+     * It is expected that all entries be valid symbols or symbolic expressions.
+     *
+     * @tparam SymbolicType Any symbolic type that is understood by the
+     *         add_to_symbol_map() functions. This includes individual
+     *         Expression, std::vector<Expression>, as well as
+     *         Tensors and SymmetricTensors of Expressions.
+     */
+    template <typename SymbolicType>
+    types::substitution_map
+    make_symbol_map(const SymbolicType &symbol);
+
+    /**
+     * Return a symbolic map that has the entry keys given by @p symbol and all
+     * @p other_symbols. It is expected that all entries be valid symbols or
+     * symbolic expressions.
+     *
+     * With this function it is possible to construct a symbolic map from
+     * different types. An example may be as follows:
+     *
+     * @code
+     *   const types::substitution_map symbol_map
+     *     = make_symbol_map(
+     *         Expression(...),
+     *         Tensor<1,dim,Expression>(...),
+     *         SymmetricTensor<2,dim,Expression>(...));
+     * @endcode
+     *
+     * @tparam SymbolicType Any symbolic type that is understood by the
+     *         add_to_symbol_map() functions. This includes individual
+     *         Expression, std::vector<Expression>, as well as
+     *         Tensors and SymmetricTensors of Expressions.
+     */
+    template <typename SymbolicType, typename... Args>
+    types::substitution_map
+    make_symbol_map(const SymbolicType &symbol, const Args &... other_symbols);
+
+    /**
+     * A convenience function for an adding empty entry, with the key value
```

Comment:
```suggestion
     * A convenience function for an adding an empty entry, with the key value
```
What does "empty" mean in this context? It's a map, so what is the value?